### DISCUSSION TITLE
Missing \endcsname inserted.

### DISCUSSION THREAD

**Discussion by aalperakis** [TeXworks/texworks#1027] [Q&A]
Hi everybody,

I am working on creating 3D PDF from u3d files. The default code is :

\documentclass[a4paper]{article}

\usepackage[3D]{movie15}

\usepackage{hyperref}

\usepackage[english]{babel}

\begin{document}

\includemovie[

poster,

toolbar, %same as \`controls'

label=dene.u3d,

text=(dene.u3d),

3Daac=60.000000, 3Droll=0.000000, 3Dc2c=0.001530 -0.061040 0.009500, 3Droo=0.061794, 3Dcoo=0.001530 0.010000 -0.009500,

3Dlights=CAD,
]{\linewidth}{\linewidth}{dene.u3d}

\end{document}

I am getting this error :

! Missing \endcsname inserted.

<to be read again>

\message

l.13 ]{\linewidth}{\linewidth}{dene.u3d}

^^M

?

How can I over come this problem?

Best Regards

Alper

**Comment by stloeffler** [comment]
According to https://tex.stackexchange.com/a/572338, movie15 is obsolete and should be replaced by media9. So something like this might work:
```latex
\documentclass[a4paper]{article}
\usepackage{media9}
\begin{document}
\includemedia[
width=\linewidth,height=\linewidth,
3Dtoolbar,
3Daac=60.000000, 3Droll=0.000000, 3Dc2c=0.001530 -0.061040 0.009500, 3Droo=0.061794, 3Dcoo=0.001530 0.010000 -0.009500,
3Dlights=CAD
]{Click Me}{dene.u3d}
\end{document}
```
